src/WorkflowStep.spec.ts

```diff
@@ -0,0 +1,395 @@
+    });
+  });
+
+  describe('step utility functions', () => {
+    it('configure should call views.open', async () => {
+      const fakeEditArgs = (createFakeStepEditAction() as unknown) as AllWorkflowStepMiddlewareArgs;
+
+      const fakeClient = { views: { open: sinon.spy() } };
+      fakeEditArgs.client = (fakeClient as unknown) as WebClient;
+
+      const { prepareStepArgs } = await importWorkflowStep();
+      // casting to returned type because prepareStepArgs isn't built to do so
+      const editStepArgs = prepareStepArgs(fakeEditArgs) as AllWorkflowStepMiddlewareArgs<
+        WorkflowStepEditMiddlewareArgs
+      >;
+
+      await editStepArgs.configure({ blocks: [] });
+
+      assert(fakeClient.views.open.called);
+    });
+
+    it('update should call workflows.updateStep', async () => {
+      const fakeSaveArgs = (createFakeStepSaveEvent() as unknown) as AllWorkflowStepMiddlewareArgs;
+
+      const fakeClient = { workflows: { updateStep: sinon.spy() } };
+      fakeSaveArgs.client = (fakeClient as unknown) as WebClient;
+
+      const { prepareStepArgs } = await importWorkflowStep();
+      // casting to returned type because prepareStepArgs isn't built to do so
+      const saveStepArgs = prepareStepArgs(fakeSaveArgs) as AllWorkflowStepMiddlewareArgs<
+        WorkflowStepSaveMiddlewareArgs
+      >;
+
+      await saveStepArgs.update();
+
+      assert(fakeClient.workflows.updateStep.called);
+    });
+
+    it('complete should call workflows.stepCompleted', async () => {
+      const fakeExecuteArgs = (createFakeStepExecuteEvent() as unknown) as SlackWorkflowStepMiddlewareArgs &
+        AllMiddlewareArgs;
+
+      const fakeClient = { workflows: { stepCompleted: sinon.spy() } };
+      fakeExecuteArgs.client = (fakeClient as unknown) as WebClient;
+
+      const { prepareStepArgs } = await importWorkflowStep();
+      // casting to returned type because prepareStepArgs isn't built to do so
+      const executeStepArgs = prepareStepArgs(fakeExecuteArgs) as AllWorkflowStepMiddlewareArgs<
+        WorkflowStepExecuteMiddlewareArgs
+      >;
+
+      await executeStepArgs.complete();
+
+      assert(fakeClient.workflows.stepCompleted.called);
+    });
+
+    it('fail should call workflows.stepFailed', async () => {
+      const fakeExecuteArgs = (createFakeStepExecuteEvent() as unknown) as SlackWorkflowStepMiddlewareArgs &
+        AllMiddlewareArgs;
+
+      const fakeClient = { workflows: { stepFailed: sinon.spy() } };
+      fakeExecuteArgs.client = (fakeClient as unknown) as WebClient;
+
+      const { prepareStepArgs } = await importWorkflowStep();
+      // casting to returned type because prepareStepArgs isn't built to do so
+      const executeStepArgs = prepareStepArgs(fakeExecuteArgs) as AllWorkflowStepMiddlewareArgs<
+        WorkflowStepExecuteMiddlewareArgs
+      >;
+
+      await executeStepArgs.fail({ error: { message: 'Failed' } });
+
+      assert(fakeClient.workflows.stepFailed.called);
```

Comment:
similar to above, is there anything else we want to verify besides the fact that it was called?